### ISSUE TITLE
Decimal separator support for multiple languages

### ISSUE THREAD

**Issue by sadeiasturias** [open-sdg/open-sdg#991]
Some languages, like German or Spanish, use a comma as a decimal separator instead of a point.
Would it be possible to show numbers in this way in tables and graphics? E.g. `13,2` instead of `13.2`.

We have tried parsing numbers by means of a function in `custom.js` (inspired by the ONS site @LucyGwilliamAdmin):

```
opensdg.dataRounding = function(value) {
  if (value == null) {
    return value
  }
  else {
    return value.toFixed(1).replace('.', ',')
    // returns a character (e.g. "13.0")
    // for mose reason, "0.0" is still shown as "0" in tables
  }
};
```

Then values are shown as wanted in tables, but graphics stop to work.

**Comment by MoBosse** [comment]
Yes, Germany would welcome that too. 
In fact our national statistical office even has an in-house corporate design that says that we should use a whitespace as thousand-seperator, so a general configurability would be great!
![image](https://user-images.githubusercontent.com/50657629/100888879-dc523780-34b6-11eb-8a51-d8c7ee95d31b.png)

**Comment by brockfanning** [comment]
I was reading up on [toLocaleString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString) and it sounds promising here. I think we should be able to run all the chart/table values through this function.

**Comment by sadeiasturias** [comment]
Thanks for the feedback @brockfanning. Indeed it might be that `toLocaleString` takes care of the number of decimals, too. Replacing in the example above

```
value.toFixed(1).replace('.', ',')
```
with the sentence

```
value.toLocaleString('es-ES', {maximumFractionDigits: 1, minimumFractionDigits: 1})
```

seems to have the same effect (e.g. transforming `13045.302` into `13.045,3`). Problem is, as before, that this naive implementation just in `custom.js` causes all charts to stop working; possibly because using comma as decimal separator is incompatible with the charting library.

> EDIT: Though I'm not sure how configurable that would be - re: using white-spaces as the thousands separator.

It doesn't seem to be very configurable, as far as I could see. In this particular case (white space fot thousands, comma for decimal), the french `fr-FR` setting does the trick @MoBosse .

**Comment by brockfanning** [comment]
@sadeiasturias @MoBosse It sounds like this would need to be an alteration of the values just before presentation, which means hooking into the Chart.js configuration (for charts) and the DataTables configuration (for tables). We already have a way to do this for charts, and we have a PR that's open now for tables (#995).

Here's an example of how you could do this right now, for charts:

```
function convertDataValue(value) {
    return value.toLocaleString('es-ES', {maximumFractionDigits: 1, minimumFractionDigits: 1});
}
opensdg.chartConfigAlter(function(config, info) {
    var overrides = {
        options: {
            tooltips: {
                callbacks: {
                    // Override the labels of tooltips.
                    // See https://www.chartjs.org/docs/latest/configuration/tooltip.html#label-callback
                    label: function(tooltipItems, data) {
                        return tooltipItems.label + ': ' + convertDataValue(tooltipItems.yLabel);
                    }
                }
            },
            scales: {
                yAxes: [{
                    ticks: {
                        // Override the labels on the Y axis.
                        // See https://www.chartjs.org/docs/latest/axes/labelling.html#creating-custom-tick-formats
                        callback: function(value) {
                            return convertDataValue(value);
                        }
                    }
                }]
            }
        }
    }
    // Add these overrides onto the normal config.
    $.extend(true, config, overrides);
});
```

But I suspect we may need something more simple and direct, like an `opensdg.dataDisplay` hook, or something like that, since this is going to be commonly requested.

**Comment by brockfanning** [comment]
@sadeiasturias Just to follow up now that this has been released in version 1.3: The site configuration setting is described here: https://open-sdg.readthedocs.io/en/stable/configuration/#decimal_separator

@MoBosse There is also now a "hook" where you can do more advanced customization of the data display. Here is some documentation of that: https://open-sdg.readthedocs.io/en/stable/customisation/#altering-data-values-before-displaying
Note that (importantly) that hook can return a string.
